Build v1.6.0 docs with 0.7 vision using new CI changes

jlin27 opened the pull request:
For Pytorch 1.6 release, vision docs were built off of master. Rebuilding the docs to build from a release/0.7.

This PR is using the `pytorch_doc_push` and `pytorch_python_doc_build` jobs to build the docs and push them to pytorch.github.io

I forked `release/1.6` and made the following changes: 
* Added the recent changes from .circle related to docs
* Added in newest .jenkins files so that it `/.jenkins/pytorch/common_utils.sh` in order to use the TORCH_COMMIT hash

seemethere committed:
.circleci: Have python docs always push to site (#42552)

Summary:
Was getting an error when attempting to push to master for
pytorch/pytorch.github.io since the main branch on that repository is
actually site and not master.

Get rid of the loop too since the loop wasn't going to work with a
conditional and conditionals on a two variable loop just isn't worth the
readability concerns

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Pull Request resolved: https://github.com/pytorch/pytorch/pull/42552

Reviewed By: malfet

Differential Revision: D22929503

Pulled By: seemethere

fbshipit-source-id: acdd26b86718304eac9dcfc81761de0b3e609004

Commit:
[Don't merge] v1.6.0 with 0.7 vision

Commit:
Add in netrc for docs push and separate out docs build from push

Commit:
Correct python_doc_push_Script.sh

Commit:
Update .jenkins folder and change torchvision commit hash

jlin27 commented on .circleci/config.yml:
```diff
@@ -1319,39 +1392,30 @@ jobs:
-          else
-            export COMMAND='((echo "export BUILD_ENVIRONMENT=${BUILD_ENVIRONMENT}" && echo "export GITHUB_PYTORCHBOT_TOKEN=${GITHUB_PYTORCHBOT_TOKEN}" && echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace && cd workspace && . ./.circleci/scripts/python_doc_push_script.sh docs/master master site dry_run") | docker exec -u jenkins -i "$id" bash) 2>&1'
-          fi
+          export COMMAND='((echo "export BUILD_ENVIRONMENT=${BUILD_ENVIRONMENT}" && echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace && cd workspace && . ./.circleci/scripts/python_doc_push_script.sh docs/master master site") | docker exec -u jenkins -i "$id" bash) 2>&1'
```
@seemethere - Will I be able to set this line to build from a branch other than master (e.g. release/1.6)  and push to a branch other than site (e.g. site-v1.6.0)?

jlin27 replied:
`export COMMAND='((echo "export BUILD_ENVIRONMENT=${BUILD_ENVIRONMENT}" && echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace && cd workspace && . ./.circleci/scripts/python_doc_push_script.sh docs/stable 1.6.0 site-v1.6.0") | docker exec -u jenkins -i "$id" bash) 2>&1'`

seemethere replied:
Oh I see, it's failing now since the docker image that it's based on probably got reaped, will push an update to the docker image to a more recent tag.

seemethere committed:
DO NOT MERGE, bump docker tag, remove a bunch of build jobs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
DO NOT MERGE, removing nightly filter

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
comment out custom backend test for now

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
update docker image tag for docs builds

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
switch to site-v1.6.0

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
Don't build and push things we don't need

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Commit:
Change from push to master instead to push site-v1.6.0

Configure pytorch_python_doc_build to push to `site-v1.6.0` instead of `master`

Commit:
Update to push to site-v1.6.0 instead of master

zou3519 commented on .circleci/cimodel/data/pytorch_build_definitions.py:
```diff
@@ -158,8 +184,19 @@ def gen_dependent_configs(xenial_parent_config):
+            parent_build=xenial_parent_config
+        )
+    )
+    configs.append(
+        DocPushConf(
+            "pytorch_python_doc_push",
+            parent_build="pytorch_python_doc_build",
+            branch="site-v1.6.0",
+        )
+    )
```
Just confirming -- is this what does the actual push to `site-v1.6.0`?

zou3519 replied:
e.g., when we release 1.7.0 documentation, do we have to add this line to that PR?

jlin27 commented:
@seemethere @zou3519 
Do you see any issues if we merge this PR into `release/1.6`? We'd like to setup the build in the 1.6 branch so it's possible to re-build stable docs at anytime without needing to make these changes again.
